commerce/domain/shipping/splitshipment/SplitShipmentSample.cls

```diff
@@ -0,0 +1,198 @@
+            if (deliveryAddress.getCountry() != null) {
+                cartDeliveryGroup.setDeliverToCountry(deliveryAddress.getCountry());
+            }
+            if (deliveryAddress.getStateCode() != null) {
+                cartDeliveryGroup.setDeliverToStateCode(deliveryAddress.getStateCode());
+            }
+            if (deliveryAddress.getCountryCode() != null) {
+                cartDeliveryGroup.setDeliverToCountryCode(deliveryAddress.getCountryCode());
+            }
+            if (deliveryAddress.getLatitude() != null) {
+                cartDeliveryGroup.setDeliverToLatitude(deliveryAddress.getLatitude());
+            }
+            if (deliveryAddress.getLongitude() != null) {
+                cartDeliveryGroup.setDeliverToLongitude(deliveryAddress.getLongitude());
+            }
+            if (deliveryAddress.getGeocodeAccuracy() != null) {
+                cartDeliveryGroup.setDeliverToGeocodeAccuracy(deliveryAddress.getGeocodeAccuracy());
+            }
+        }
+
+        if (itemArrange.getDeliverToName() != null) {
+            cartDeliveryGroup.setDeliverToName(itemArrange.getDeliverToName());
+        }
+        if (itemArrange.getDeliverToFirstName() != null) {
+            cartDeliveryGroup.setDeliverToFirstName(itemArrange.getDeliverToFirstName());
+        }
+        if (itemArrange.getDeliverToLastName() != null) {
+            cartDeliveryGroup.setDeliverToLastName(itemArrange.getDeliverToLastName());
+        }
+        if (itemArrange.getDeliverToCompanyName() != null) {
+            cartDeliveryGroup.setCompanyName(itemArrange.getDeliverToCompanyName());
+        }
+
+        cart.getCartDeliveryGroups().add(cartDeliveryGroup);
+        return cartDeliveryGroup;
+    }
+
+    /**
+     * @description Sets the quantity and associates the target Cart Delivery Group for a Cart Item, if it is provided. 
+     * If a Cart Item is not provided, then a new one is created with the provided quantity and Cart Delivery Group.
+     * 
+     * @param cart Representation of a `WebCart` record
+     * @param targetCartDeliveryGroup The Cart Delivery Group to associate with the Cart Item
+     * @param itemArrange Holds details of how to arrange an item between delivery groups
+     */
+    private void updateOrCreateCartItemToCartDeliveryGroup(CartExtension.Cart cart, CartExtension.CartDeliveryGroup targetCartDeliveryGroup, CartExtension.ItemArrange itemArrange) {
+        if (itemArrange.getCartItemId() == null) {
+            CartExtension.CartItem cartItem = new CartExtension.CartItem(CartExtension.SalesItemTypeEnum.PRODUCT, targetCartDeliveryGroup, 'My Cart Item');
+            cartItem.setProduct2Id(itemArrange.getProductId());
+            cartItem.setQuantity(itemArrange.getQuantity());
+            cart.getCartItems().add(cartItem);
+        } else {
+            Iterator<CartExtension.CartItem> cartItemsIterator = cart.getCartItems().iterator();
+            while (cartItemsIterator.hasNext()) {
```

Comment:
Isn't this taken care of by the allocation api?  Seems exceptionally complicated to replicate that logic here.  Eg: merging would be a nightmare.

Reply:
This is in the default implementation of the Item Arrangement API, so technically Customers should also have the ability to do this? We can use a simpler example though instead of following the default implementation.